sys/net/gnrc/network_layer/ipv6/nib/_nib-arsm.c

```diff
@@ -455,6 +458,13 @@ void _set_nud_state(gnrc_netif_t *netif, _nib_onl_entry_t *nce,
 #else   /* CONFIG_GNRC_IPV6_NIB_ROUTER */
     (void)netif;
 #endif  /* CONFIG_GNRC_IPV6_NIB_ROUTER */
+#if IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_QUEUE_PKT)
+    if (was_reachable && !_is_reachable(nce) && nce->pktqueue_len > 0) {
+        _evtimer_add(nce, GNRC_IPV6_NIB_FLUSH_PCK_QUEUE,
+                     &nce->flush_queue_timeout,
+                     CONFIG_GNRC_IPV6_NIB_QUEUE_PKT_LINGER_MS);
+    }
```

Comment:
Not sure but think about it:
If the neighbor becomes reachable again suddenly, should the timer to flush the queue be canceled.

I guess,

```C
void _nbr_flush_pktqueue(_nib_onl_entry_t *node)
{
    if (_is_reachable(node)) {
        return;
    }
[...]
```
is supposed to catch that case, right?

Reply:
Exactly. There is no common entry point for setting a neighbor reachable, so that we can cancel the the timer there. Also `_is_reachable() == true` doesn't require the neighbor to be in REACHABLE state, but to be neither in INCOMPLETE nor UNREACHABLE. We probably don't want to discard the packets in that case.

Reply:
regarding memory safety, the timer is canceled before the neighbor entry goes out of scope anyway.